Sources/Features/Breadcrumb/BacktraceBreadcrumbFile.swift

```diff
@@ -0,0 +1,119 @@
+            BacktraceLogger.warning("\(breadcrumbSettings.maxQueueFileSizeBytes) is smaller than the minimum of " +
+                                    "\(BacktraceBreadcrumbFile.minimumQueueFileSizeBytes)" +
+                                    ", ignoring value and overriding with minimum.")
+            self.maxQueueFileSizeBytes = BacktraceBreadcrumbFile.minimumQueueFileSizeBytes
+        } else {
+            self.maxQueueFileSizeBytes = breadcrumbSettings.maxQueueFileSizeBytes
+        }
+
+        super.init()
+    }
+
+    func addBreadcrumb(_ breadcrumb: [String: Any]) -> Bool {
+        do {
+            // Serialize breadcrumb: [String: Any] into Data
+            let breadcrumbJsonData = try JSONSerialization.data(withJSONObject: breadcrumb)
+            // Serialize Data into a JSON string
+            guard let breadcrumbJsonString = String(data: breadcrumbJsonData, encoding: .utf8) else {
+                BacktraceLogger.warning("Error when converting breadcrumb to string")
+                return false
+            }
+            // Calculate the size of the breadcrumb and add it to queue
+            let breadcrumbSize = breadcrumbJsonData.count
+            // Check if breadcrumb size is larger than the maximum specified
+            if breadcrumbSize > maximumIndividualBreadcrumbSize {
+                BacktraceLogger.warning(
+                    "Discarding breadcrumb that was larger than the maximum specified (\(maximumIndividualBreadcrumbSize).")
+                return false
+            }
+            // Store breadcrumb Json String and size in Dictionary [String : Any]
+            let queueBreadcrumb = ["breadcrumbJson": breadcrumbJsonString, "size": breadcrumbSize] as [String : Any]
+            // Queue breacrumb
+            queue.enqueue(queueBreadcrumb)
+            // Iterate over the queue from newest to oldest breadcrumb and build an array of encoded strings
+            _ = dispatchQueue.sync {
+                let queuedBreadcrumbs = queue.allElements()
+                var breadcrumbsArray = [String]()
+                var size = 0
+                for index in (0..<queue.count).reversed() {
+                    guard let queueBreadcrumb = queuedBreadcrumbs[index] as? [String: Any] else {
+                        BacktraceLogger.warning("Error when fetching breacrumbs from queue")
+                        return false
+                    }
+                    guard let breadcrumbSize = queueBreadcrumb["size"] as? Int else {
+                        BacktraceLogger.warning("Error when adding breadcrumbSize to array")
+                        return false
+                    }
+                    // Pop last element if size is greater than maxQueueFileSizeBytes
+                    if size + breadcrumbSize > maxQueueFileSizeBytes && !queue.isEmpty {
+                        while (index != 0) {
+                            _ = queue.pop(at: index)
+                        }
+                    } else {
```

Comment:
if you add break to the if statement above, you don't need to have else and one more identation